### PR TITLE
Add zip file creation for signatures

### PR DESCRIPTION
**Pull request by adamnitti**
## overall purpose
adds zip file packaging for generated signature html files

### CODE REVIEW

File: csv-bulk-signature/src/email-sig-template.html [hunk 1]
```diff
@@ -67,7 +67,7 @@
               <tbody>
                 <tr>
                   <td class="blue-bold-text">
-                    {{fullName}}{{#if title}}, {{credentials}}{{/if}}
+                    {{fullName}}{{#if credentials}}, {{credentials}}{{/if}}
```
Comment (adamnitti) [inline csv-bulk-signature/src/email-sig-template.html:70]: correcting handlebars variable

File: csv-bulk-signature/src/main.ts [hunk 2]
```diff
@@ -107,10 +109,10 @@ function generateSignatures(
         "for missing required fields (scroll up to see list)"
       );
     });
+  zipUpFile();
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:112]: moved this function call to outside of `.on` block. was resulting in EventEmitter errors due to piping multiple readstreams into the same zip stream

File: csv-bulk-signature/src/main.ts [hunk 3]
```diff
@@ -173,3 +175,24 @@ function checkHeadersToBeSame() {
...
+
+async function zipUpFile() {
+  const folderPath = "./dist";
+  const files = fs.readdirSync(folderPath);
+  const filesHtm = files.filter((file) => file.endsWith(".htm"));
+
+  const zip = new jszip();
+
+  filesHtm.forEach((file) => {
+    try {
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:187]: added error checking

File: csv-bulk-signature/src/main.ts [hunk 4]
```diff
@@ -173,3 +175,24 @@ function checkHeadersToBeSame() {
       counter = 1;
     });
 }
+
+async function zipUpFile() {
+  const folderPath = "./dist";
+  const files = fs.readdirSync(folderPath);
+  const filesHtm = files.filter((file) => file.endsWith(".htm"));
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:182]: refactored filtered files

File: csv-bulk-signature/src/main.ts [hunk 5]
```diff
@@ -173,3 +175,24 @@ function checkHeadersToBeSame() {
...
+      const filePath = `${folderPath}/${file}`;
+      const fileContents = fs.readFileSync(filePath);
+      zip.file(file, fileContents);
+    } catch (error) {
+      console.error(`Error adding file ${file}: ${error.message}`);
+    }
+  });
+
+  const buffer = await zip.generateAsync({ type: "nodebuffer" });
+  fs.writeFileSync("signature-templates.zip", buffer);
```
Comment (chancesmith) [inline csv-bulk-signature/src/main.ts]: To keep track of signature builds, what if we remove `templates` and replace with a datetime?
`signatures-22-12-27-11:30.zip`

**Review by chancesmith — APPROVED** [review]
Small tweak to zip name.

File: csv-bulk-signature/src/main.ts (hunk 5 above)
Comment (adamnitti) [reply csv-bulk-signature/src/main.ts]: 👍

File: csv-bulk-signature/src/main.ts [hunk 6]
```diff
@@ -173,3 +175,24 @@ function checkHeadersToBeSame() {
...
+      const filePath = `${folderPath}/${file}`;
+      const fileContents = fs.readFileSync(filePath);
+      zip.file(file, fileContents);
+    } catch (error) {
+      console.error(`Error adding file ${file}: ${error.message}`);
+    }
+  });
+
+  const buffer = await zip.generateAsync({ type: "nodebuffer" });
+  fs.writeFileSync(`signatures-${new Date().toDateString()}.zip`, buffer);
```
Comment (adamnitti) [inline csv-bulk-signature/src/main.ts:197]: adding date stamp to zip filename

**Pull request merged**